src/poller/utils.rs

```diff
@@ -231,23 +288,29 @@ pub async fn save_url_to_internet_archive_urls(
     url: &str,
     from_table: &str,
     from_table_id: i32,
-    pool: &PgPool) {
-    if should_insert_url_to_internet_archive_urls(url, pool).await.expect("Error: ") {
-        let query = r#"
+    pool: &PgPool,
+) {
+    match should_insert_url_to_internet_archive_urls(url, pool).await {
+        Ok(can_insert) => {
+            if can_insert {
+                let query = r#"
         INSERT INTO external_url_archiver.internet_archive_urls (url, from_table, from_table_id, retry_count, is_saved)
          VALUES ($1, $2, $3, 0, false)"#;
-        sqlx::query(query)
-            .bind(url)
-            .bind(from_table)
-            .bind(from_table_id)
-            .execute(pool)
-            .await
-            .unwrap();
-    } else {
-        return;
+                sqlx::query(query)
+                    .bind(url)
+                    .bind(from_table)
+                    .bind(from_table_id)
+                    .execute(pool)
+                    .await
+                    .unwrap();
+            }
+        }
+        Err(e) => {
+            eprintln!("Error saving {} into internet_archive_urls: {}", url, e)
+        }
     }
 }
```

Comment:
@RustyNova016 updated the logic here

Reply:
Looks good to me. 👍

Reply:
@RustyNova016 btw, should I try to `Return<(), Error>` in this case?
I think in general, in functions related to db access, I should do the same. Thoughts?

Reply:
Here's some rules of thumb I try to follows:
- Errors that I don't want to deal here (ex: not enough context about the caller)? Return the error
- Got an error that you know you can handle here and the caller isn't a better place to handle? Handle the error state here
- Got an error that you know is impossible to happen (like inserting in a Vec then getting the first element)? Use `.except()` and provide an explanation of why this isn't an issue, and if that ever happens, some clues as to why it may happen.

In your case, just return `Result<>`. But this isn't a change in a vacuum. IMO removing all those unwraps and setting function returns should be it's own PR, but I'm not your project manager so ehh

Reply:
Thanks, I will come up with a cleanup and refactoring PRs for this.
I didn't really focus much on doing things idiomatic way, rather exploring different ways to do them.
Thanks to your PR, I am getting a lot of sense of clean code, and will try to incorporate the coding style in future contributions ✅.